pkg/lib/cockpit.js

```diff
@@ -1483,17 +1483,27 @@ function factory() {
             });
     };
 
-    function format_units(number, suffixes, factor, options) {
-        // backwards compat: "options" argument position used to be a boolean flag "separate"
-        if (!is_object(options))
-            options = { separate: options };
+    function format_units(suffixes, number, second_arg, third_arg) {
+        let options = second_arg;
 
+        /* compat API: we used to accept 'factor' as a separate second arg */
+        if (third_arg || (second_arg && !is_object(second_arg))) {
+            options = third_arg;
+
+            // double backwards compat: "options" argument position used to be a boolean flag "separate"
+            if (!is_object(options))
+                options = { separate: options };
+
+            options.factor = second_arg;
```

Comment:
This looks unclean: If options is already an object, it is passed in from the caller. This will modify that passed-in argument. So as an intermediate step, this is safer: `options = {...options, factor: second_arg}` (i.e. make a copy).

But even more so this is a field which isn't documented or covered by types, so doesn't this raise some ts alarm?

WDYT about something like
```js
if (second_arg === 1024)
    options = {...options, base2: true }
else
   console.warn("format_units(): Ignoring invalid factor value", second_arg);
```
?

Reply:
Thanks.  This is a good suggestion and I'm surprised I was sloppy about this.